WangscapeGui/mainwindow.h

```diff
@@ -0,0 +1,52 @@
+#ifndef MAINWINDOW_H
+#define MAINWINDOW_H
+
+#include <memory>
+
+#include <QGraphicsScene>
+#include <QMainWindow>
+
+#include <OptionsManager.h>
+#include <tilegen/TilesetGenerator.h>
+#include <Options.h>
+
+#include "optionseditor.h"
+
+namespace Ui {
+class MainWindow;
+}
+
+class MainWindow : public QMainWindow
+{
+    Q_OBJECT
+
+public:
+    explicit MainWindow(QWidget *parent = 0);
+    ~MainWindow();
+
+private slots:
+    void generate();
+    void saveOutput();
+    void loadOptionsFromFile();
+    void displayTilesetPreview(const QString& name=QString());
+    void openOptionsEditor();
+    void resetOptions();
+
+private:
+    void initializePreviewArea();
+    QImage convertSfImageToQImage(const sf::Image &source_image);
+    void resetTilesetGenerator();
+
+    Ui::MainWindow *mUi;
+    QGraphicsScene *mScene;
+    // TODO(hryniuk): create a struct for this map's elements
+    std::map<std::string, std::pair<std::string, QImage>> mPreviewImages;
+    QString mOptionsFilePath;
+    std::unique_ptr<OptionsManager> mOptionsManager;
+    Options mOriginalOptions;
+    Options mOptions;
+    OptionsEditor* mOptionsEditor;
```

Comment:
Is it possible to use smart pointers for `mUi`, `mScene`, and `mOptionsEditor`?

Reply:
As far as I know yes, but it's **not recommended**, cause Qt has it's own memory management system:
https://stackoverflow.com/questions/34433435/why-dont-the-official-qt-examples-and-tutorials-use-smart-pointers

I'm not really sure about mUi (from what I've read I conclude that it's handled by Qt too), but others are QObjects, thus they are properly released.